src/azure-cli-core/tests/test_progress.py

```diff
@@ -0,0 +1,133 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import unittest
+
+import azure.cli.core.commands.progress as progress
+
+
+class MockOutstream(progress._ProgressViewBase):
+    """ mock outstream for testing """
+    def __init__(self, p_type=progress.ProgressType.Indeterminate):
+        self.string = ''
+        self.type = p_type
+
+    def write(self, message):
+        self.string = message
+
+    def get_type(self):
+        return self.type
+
+    def flush(self):
+        return self.string
+
+
+class DetMockOutstream(progress._ProgressViewBase):
+    """ mock outstream for testing """
+    def __init__(self, p_type=progress.ProgressType.Determinate):
+        self.string = ''
+        self.type = p_type
+
+    def write(self, message):
+        self.string = message
+
+    def get_type(self):
+        return self.type
+
+    def flush(self):
+        return self.string
+
+
+class TestProgress(unittest.TestCase):  # pylint: disable=too-many-public-methods
+    """ test the progress reporting """
+
+    def test_det_model(self):
```

Comment:
I suggest renaming all these test method names to be more explicit about the module they are testing. eg. `test_progress_indicator_det_mode;`